internal/pkg/policy/secret_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !integration
```

Comment:
I don't understand though we we have to exclude unit tests like this explicitly, as integration tests are tagged also.
Are we missing something?
the Makefile has this command to tigger integration tests:
`go test -v -tags=integration -count=1 -race -p 1 ./...`

Reply:
I guess that the idea is to don't run unit tests when running integration tests, so the same tests are not executed at two different stages of testing.

Reply:
Yes, I mean shouldn't it be enough to mark integration tests with `//go:build integration`? I think it's easy to forget when adding new unit tests to add the negated condition, and I'm trying to understand why it is needed.

Reply:
Files without any build tag are always included, so `-tags=integration` wouldn't discard unit test files if they don't have the negated tag. So we need `!integration` for test files that we don't want to execute during integration tests.

I agree that this is easy to forget. Tough I wouldn't see as a great problem to execute some fast unit tests during integration test execution.

If we strongly want to have their execution separated, a possible strategy to don't forget it is to have only unit tests in packages, and move the integration tests to their own package/s, in a similar fashion to what we do for [e2e tests](https://github.com/elastic/fleet-server/tree/main/testing). This also help to validate interfaces. 
Or we could have some check in CI to ensure that all test files have tags.